packages/@aws-cdk/aws-eks/lib/oidc-provider.ts

```diff
@@ -0,0 +1,66 @@
+import * as iam from '@aws-cdk/aws-iam';
+import { Construct } from 'constructs';
+
+/**
+ * Initialization properties for `OpenIdConnectProvider`.
+ */
+export interface OpenIdConnectProviderProps {
+  /**
+   * The URL of the identity provider. The URL must begin with https:// and
+   * should correspond to the iss claim in the provider's OpenID Connect ID
+   * tokens. Per the OIDC standard, path components are allowed but query
+   * parameters are not. Typically the URL consists of only a hostname, like
+   * https://server.example.org or https://example.com.
+   *
+   * You can find your OIDC Issuer URL by:
+   * aws eks describe-cluster --name %cluster_name% --query "cluster.identity.oidc.issuer" --output text
+   */
+  readonly url: string;
+
+  /**
+   * Physical name of provider
+   *
+   * @default undefined you can leave it empty so the name will be auto-generated
+   */
+  readonly customResourceName?: string;
```

Comment:
Not used, and not really needed. We can remove it.

Reply:
Tried to remove it but the `cdk-build` returns:
`error: [awslint:props-physical-name:@aws-cdk/aws-eks.OpenIdConnectProviderProps] Every Resource must have a single physical name construction property, with a name that is an ending substring of <cfnResource>Name`